core/src/util.jl

```diff
@@ -417,9 +417,21 @@ function get_all_priorities(db::DB, config::Config)::Vector{Int32}
         FlowDemandStaticV1,
         FlowDemandTimeV1,
     ]
-        union!(priorities, load_structvector(db, config, type).priority)
+        if valid_priorities(
+            load_structvector(db, config, type).priority,
+            config.allocation.use_allocation,
+        )
+            union!(priorities, load_structvector(db, config, type).priority)
+        else
+            valid_p = false
+            @error "Missing priority parameter(s) for a $type node in the allocation problem."
```

Comment:
I don't want to expose users to the internal name `Ribasim.UserDemandStaticV1`. I think you can use the `nodetype` function to get an actual name.